Source/ASDisplayNode.mm

```diff
@@ -2281,6 +2281,15 @@ - (void)removeFromSupernode
 {
   ASDisplayNodeLogEvent(self, @"removeFromSupernode with automaticallyManagesSubnodes: %@",
                         self.automaticallyManagesSubnodes ? @"YES" : @"NO");
+
+#if ASDISPLAYNODE_ASSERTIONS_ENABLED
+  __instanceLock__.lock();
+    __weak ASDisplayNode *supernode = _supernode;
+  __instanceLock__.unlock();
+
+  ASDisplayNodeAssert(! supernode.automaticallyManagesSubnodes, @"Should not manually remove node %@ from its supernode %@ which has automaticallyManagesSubnodes enabled. Please talk to Huy if you're hitting this in closeup!", self, supernode);
```

Comment:
We should maybe not add your name in here. What about either using the github name or say something like: "Open an issue etc."

Reply:
The last sentence is meant for P engineers only :)